src/protocol/libp2p/kademlia/query/find_node.rs

```diff
@@ -131,9 +131,18 @@ impl<T: Clone + Into<Vec<u8>>> FindNodeContext<T> {
             self.responses.insert(distance, peer);
         } else {
             // Update the furthest peer if this response is closer.
-            if let Some(mut entry) = self.responses.last_entry() {
-                if entry.key() > &distance {
-                    entry.insert(peer);
+            // Find the furthest distance.
+            let furthest_distance =
+                self.responses.last_entry().map(|entry| *entry.key()).unwrap_or(distance);
+
+            // The response received from the peer is closer than the furthest response.
+            if distance < furthest_distance {
+                // Update the entries only if the distance is not already present.
```

Comment:
What is the drawback of overwriting the entry with this distance? I.e., there should be exactly one peer with such distance as it uniquely identifies the hash, so it should be safe to overwrite without checking.

Reply:
I think I followed the libp2p here. However, I believe we can overwrite the entries without checking here, and we'll have a "fresher" report in the end. Will do that, thanks!